dataRetriever/factory/requestersContainer/args.go

```diff
@@ -13,8 +13,8 @@ import (
 type FactoryArgs struct {
 	RequesterConfig                 config.RequesterConfig
 	ShardCoordinator                sharding.Coordinator
-	MainMessenger                   dataRetriever.TopicMessageHandler
-	FullArchiveMessenger            dataRetriever.TopicMessageHandler
+	MainMessenger                   p2p.Messenger
```

Comment:
We should evaluate that we can get rid of providing the messenger on the resolvers as the messenger is now provided on each data request. (next PR)